File: toyProblems/solved/very-easy/final-value-of-variable-after-performing-operations.js
// Leetcode #2011
// https://leetcode.com/problems/final-value-of-variable-after-performing-operations/

var finalValueAfterOperations = function(operations) {
    return operations.reduce((total, operation) => {
        return operation.indexOf('-') > -1 ? total - 1 : total + 1; 
    }, 0)
};

File: Java/toyProblems/very-easy/solved/final-value-of-variable-after-performing-operations.java
// Leetcode 2011
// https://leetcode.com/problems/final-value-of-variable-after-performing-operations/

// 12/30/21

class Solution {

  public int finalValueAfterOperations(String[] operations) {
    int x = 0;

    for (int i = 0; i < operations.length; i++) {
      if (operations[i].equals("--X") || operations[i].equals("X--")) {
        x = x - 1;
      } else {
        x = x + 1;
      }
    }
    return x;
  }
}
